test/e2e/logs_label_test.go

```diff
@@ -68,16 +78,25 @@ var _ = Describe("Logs Drop Labels", Label("logging"), func() {
 			verifiers.DeploymentShouldBeReady(ctx, k8sClient, types.NamespacedName{Namespace: mockNs, Name: logProducerName})
 		})
 
-		It("Should collect only labels and drop annotations", func() {
+		It("Should have logs with labels in the backend", func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(telemetryExportURL)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				g.Expect(resp).To(HaveHTTPBody(ContainLd(ContainLogRecord(
+					WithKubernetesLabels(HaveKeyWithValue("app", "logging-label-test")))),
+				))
+			}, periodic.TelemetryEventuallyTimeout, periodic.TelemetryInterval).Should(Succeed())
+		})
+
+		It("Should have no logs with annotations in the backend", func() {
 			Consistently(func(g Gomega) {
-				time.Sleep(20 * time.Second)
 				resp, err := proxyClient.Get(telemetryExportURL)
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
-				g.Expect(resp).To(HaveHTTPBody(SatisfyAll(
-					ContainLogs(WithKubernetesLabels("app", "logging-label-test")),
-					Not(ContainLogs(WithKubernetesAnnotations())),
-				)))
+				g.Expect(resp).To(HaveHTTPBody(Not(ContainLd(ContainLogRecord(
+					WithKubernetesAnnotations(Not(BeEmpty()))))),
```

Comment:
this is negated two times. would it also work to remove both `Not` matchers and check that the Annotations are empty?